sample-app/src/BaseRouter.tsx

```diff
@@ -0,0 +1,35 @@
+import { ComponentType } from 'react';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+
+interface RouteData {
+  path: string
+  page: JSX.Element
+  exact?: boolean
+}
+
+interface PageProps {
+  Layout?: ComponentType<{ page: JSX.Element }>
+  routes: RouteData[]
+}
+
+export default function BaseRouter({ Layout, routes }: PageProps) {
+  const pages = routes.map(routeData => {
+    const { path, page, exact } = routeData;
+    if (Layout) {
```

Comment:
I'm wondering if the Layout props adds much value to the router. I would expect that a router would simply render what you give it at a given route. I briefly looked into the react router and I didn't see any sort of layout concept, but it's possible I missed something there. What are some of the reasons that lead you to adding Layout to the router?

Reply:
I wanted to abstract away react-router details for the the kind of use case we would expect theme 2.0 to have, and wanted a way for a given route's page to have shared components. We had some idea of "layouts" for jambo that eventually fizzled out, but I thought made sense here